Chess/src/GameLogic.cpp

```diff
@@ -0,0 +1,192 @@
+#include "../include/GameLogic.h"
+#include "../include/Chess.h"
+#include "../include/Pawn.h"
+#include "../include/Knight.h"
+#include "../include/Bishop.h"
+#include "../include/Rock.h"
+#include "../include/Queen.h"
+#include "../include/King.h"
+#include <sstream>
+#include <string>
+
+
+using namespace std;
+
+enum behavior { Valid=41, SamePiece, MoveOutOfRange };
+GameLogic::GameLogic(Chess *chess) : chess(chess)
+{
+    PiecesArrays();
+}
+void GameLogic::PiecesArrays()
+{
+    PawnArrays();
+    BishopArrays();
+    KnightArrays();
+    RockArrays();
+    QueenArrays();
+    KingArrays();
+}
+
+int GameLogic::ProcessTheInput(const std::string& m_input)
+{
+    stringstream ss(m_input);
+    string token;
+    vector<pair<int, int>> ParsedInput;
+    while (ss >> token) {
+        pair<int, int> PlaceAsInteger = {0, 0};
+        for (char ch : token) {
+            if (isalpha(ch) && !islower(ch))
+                PlaceAsInteger.first = ((ch - 'A') * 2) + 3;

```

Comment:
All the special strings should be in consts for better readability and understanding of the code.